x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_main_template/index.tsx

```diff
@@ -97,9 +94,14 @@ export function ApmMainTemplate({
     [application?.capabilities.savedObjectsManagement.edit]
   );
 
-  const shouldBypassNoDataScreen = bypassNoDataScreenPaths.some((path) =>
-    location.pathname.includes(path)
-  );
+  const hasLogsData = serviceEntitySummary?.dataStreamTypes
+    ? serviceEntitySummary?.dataStreamTypes?.length > 0 &&
+      isLogsSignal(serviceEntitySummary.dataStreamTypes)
+    : false;
```

Comment:
```suggestion
  const hasLogsData = serviceEntitySummary?.dataStreamTypes?.length > 0
    ?  isLogsSignal(serviceEntitySummary.dataStreamTypes)
    : false;
```

Reply:
`serviceEntitySummary` can be undefined so we need to put the check first.

<img width="600" alt="Screenshot 2024-10-16 at 11 12 14" src="https://github.com/user-attachments/assets/c308e516-e309-4c95-8a8e-2d8df353e11f">

<img width="600" alt="Screenshot 2024-10-16 at 11 13 09" src="https://github.com/user-attachments/assets/9d0cb539-2e41-4aaf-973a-4e4a101ccb55">

Reply:
Alternatively:

```ts
(serviceEntitySummary?.dataStreamTypes?.length ?? 0) > 0
```